src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_pending_resumption_handler.cc

```diff
@@ -265,17 +267,43 @@ void VehicleInfoPendingResumptionHandler::HandleResumptionSubscriptionRequest(
     application_manager::AppExtension& extension,
     application_manager::Application& app) {
   SDL_LOG_AUTO_TRACE();
+  using namespace application_manager;
+
   sync_primitives::AutoLock lock(pending_resumption_lock_);
   SDL_LOG_TRACE("app id " << app.app_id());
   auto& ext = dynamic_cast<VehicleInfoAppExtension&>(extension);
 
-  const auto subscriptions = ext.PendingSubscriptions().GetData();
+  auto subscriptions = ext.PendingSubscriptions().GetData();
+  SDL_LOG_TRACE("resume subscriptions to : " << Stringify(subscriptions));
+  auto pending_request = SubscribeToFakeRequest(app.app_id(), subscriptions);
+  VehicleDataList already_subscribed_ivi;
+
+  for (auto ivi = subscriptions.begin(); ivi != subscriptions.end();) {
+    if (IsSubscribedAppExist(*ivi, application_manager_)) {
+      ext.RemovePendingSubscription(*ivi);
+      ext.subscribeToVehicleInfo(*ivi);
+      already_subscribed_ivi.insert(*ivi);
+      subscriptions.erase(ivi++);
+    } else {
+      ++ivi;
```

Comment:
@ychernysheva you are using increment in both branches. Maybe its simple to move it to `for` statement?

Reply:
@AKalinich-Luxoft I think, we cannot do this, because iterators of associative containers (like set in this case) are not invalidated. So if we delete something from set while iterating through it with `for` loop, order of items in this set may be changed in wrong way, so it may cause incorrect behavior. To deal with it we can use this approach with pre- and post-incrementing of iterator.